src/vs/editor/contrib/format/format.ts

```diff
@@ -169,27 +169,62 @@ export async function formatDocumentRangesWithProvider(
 		}
 	}
 
-	const allEdits: TextEdit[] = [];
+	const computeEdits = async (range: Range) => {
+		const rawEdits = await provider.provideDocumentRangeFormattingEdits(
+			model,
+			range,
+			model.getFormattingOptions(),
+			cts.token
+		);
+		return (await workerService.computeMoreMinimalEdits(model.uri, rawEdits)) || [];
```

Comment:
`computeMoreMinimalEdits` could be costly to compute for very large documents (at least the prettier formatter always replaces the entire document).

Maybe you should check if the token is still alive after the `provideDocumentRangeFormattingEdits` call?

Reply:
Added token check and called computeMoreMinimalEdits only at the end